src/Client.cpp

```diff
@@ -183,6 +210,11 @@ void Client::OnAfterCreated(CefRefPtr<CefBrowser> browser)
 
     CEF_REQUIRE_UI_THREAD();
 
+    if (!m_requestHandler->m_messageRouterBrowserSide) {
```

Comment:
I believe the constructor of the browser side message router needs to go in the UI thread but OnContextInitialized should work.

I have tested running multiple requests at once and this does seem to work. I will try OnContextIitialized.